api/src/main/java/org/openmrs/validator/VisitValidator.java

```diff
@@ -105,15 +109,25 @@ public void validate(Object target, Errors errors) {
 		// check attributes
 		super.validateAttributes(visit, errors, Context.getVisitService().getAllVisitAttributeTypes());
 		
-		// check start and end dates
-		if (disallowOverlappingVisits()) {
-			List<Visit> otherVisitList = Context.getVisitService().getVisitsByPatient(visit.getPatient());
-			for (Visit otherVisit : otherVisitList) {
-				if (visit.equals(otherVisit)) {
-					continue;
-				}
-				validateStartDatetime(visit, otherVisit, errors);
-				validateStopDatetime(visit, otherVisit, errors);
+		// Some tests are using non-existing patients. Skipping overlapping validation for such cases.
+		boolean nonExistingPatient = visit.getPatient() == null || visit.getPatient().getId() == null;
+		
+		// check for overlapping visits
+		if (disallowOverlappingVisits() && !nonExistingPatient) {
+			VisitSearchCriteria visitSearchCriteria = new VisitSearchCriteriaBuilder()
+				.setPatients(Collections.singletonList(visit.getPatient()))
```

Comment:
It seems like it would be helpful for the CriteriaBuilder to have a `setPatient()` handler...